godot-core/src/registry/plugin.rs

```diff
@@ -257,6 +268,28 @@ impl Struct {
         self
     }
 
+    pub fn with_singleton<T>(mut self) -> Self
```

Comment:
How does this interact with `class(tool)`? Runtime class singletons should not be instantiated in the editor.